content/docs/guides/crosswalk/kubernetes/aws/01-cloud-stacks/01-identity.md

```diff
@@ -0,0 +1,118 @@
+## Creating an IAM Role for Kubernetes Developers
+
+Create a developer role in AWS that assumes the AWS account caller.
+This role will be mapped into a limited developer role in Kubernetes RBAC.
+
+```typescript
+import * as aws from "@pulumi/aws";
+
+// Create the EKS cluster developers role.
+const devName = "devs";
+const devsIamRole = new aws.iam.Role(`${devName}-eksClusterDeveloper`, {
+    assumeRolePolicy: aws.getCallerIdentity().then(id =>
+        aws.iam.assumeRolePolicyForPrincipal({"AWS": `arn:aws:iam::${id.accountId}:root`}))
+})
+```
+
+## Creating IAM Roles for Kubernetes NodeGroups
+
+Create a node group worker role in AWS that assumes the EC2 Service, and attach
+required EKS cluster polices to the role. This role will be used by the node group in an
+instance profile during cluster configuration and creation.
+
+```typescript
+// The managed policies EKS requires of nodegroups join a cluster.
+const nodegroupManagedPolicyArns: string[] = [
+    "arn:aws:iam::aws:policy/AmazonEKSWorkerNodePolicy",
+    "arn:aws:iam::aws:policy/AmazonEKS_CNI_Policy",
+    "arn:aws:iam::aws:policy/AmazonEC2ContainerRegistryReadOnly",
+];
+
+// Create the standard node group worker role and attach the required policies.
+const stdNodegroupIamRoleName = "standardNodeGroup";
+const stdNodegroupIamRole = new aws.iam.Role(`${stdNodegroupIamRoleName}-eksClusterWorkerNode`, {
+    assumeRolePolicy: aws.iam.assumeRolePolicyForPrincipal({"Service": "ec2.amazonaws.com"})
+})
+attachPoliciesToRole(stdNodegroupIamRoleName, stdNodegroupIamRole, nodegroupManagedPolicyArns);
+
+// Create the performant node group worker role and attach the required policies.
+const perfNodegroupIamRoleName = "performanceNodeGroup";
+const perfNodegroupIamRole = new aws.iam.Role(`${perfNodegroupIamRoleName}-eksClusterWorkerNode`, {
+    assumeRolePolicy: aws.iam.assumeRolePolicyForPrincipal({"Service": "ec2.amazonaws.com"})
+})
+attachPoliciesToRole(perfNodegroupIamRoleName, perfNodegroupIamRole, nodegroupManagedPolicyArns);
+
+// Attach policies to a role.
+function attachPoliciesToRole(name: string, role: aws.iam.Role, policyArns: string[]) {
+    for (const policyArn of policyArns) {
+        new aws.iam.RolePolicyAttachment(`${name}-${policyArn.split('/')[1]}`,
+            { policyArn: policyArn, role: role },
+        );
+    }
+}
+```
+
+[iam]: https://aws.amazon.com/iam/
+[users]: https://docs.aws.amazon.com/IAM/latest/UserGuide/id_users_create.html
+[groups]: https://docs.aws.amazon.com/IAM/latest/UserGuide/id_groups.html
+[roles]: https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles.html
+[policies]: https://docs.aws.amazon.com/IAM/latest/UserGuide/access_policies.html
+[crosswalk-aws]: /docs/guides/crosswalk/aws
+[iam-users]: /docs/guides/crosswalk/aws/iam/#iam-users
+[iam-groups]: /docs/guides/crosswalk/aws/iam/#iam-groups
+[iam-roles]: /docs/guides/crosswalk/aws/iam/#iam-roles
+[iam-policies]: /docs/guides/crosswalk/aws/iam/#using-the-policydocument-interface
```

Comment:
Prefer `relref` for all of these, e.g.

```markdown
[crosswalk-aws]: {{< relref "/docs/guides/crosswalk/aws" >}}
[iam-users]: {{< relref "/docs/guides/crosswalk/aws/iam#iam-users" >}}
```

etc.

Reply:
Done